### DISCUSSION TITLE
[libusb_open failed] Communicate  failed between PC and Android TV

### DISCUSSION THREAD

**Discussion by Flykoki** [libusb/libusb#1552] [Q&A] [answered]
Hi ,
     PC use the libusb-win32 drvier ,  the PC program is use pyusb python script :

`dev = usb.core.find(idVendor=ACCESSORY_VID) ；dev.set_configuration()；cfg = dev.get_active_configuration()
`
  throw error when dev.setconfiguration is called:
```
File "D:\ProgramFiles\Python\lib\site-packages\usb\backend\libusb1.py", line 604, in _check
    raise USBError(_strerror(ret), ret, _libusb_errno[ret])
File "D:\ProgramFiles\Python\lib\site-packages\usb\core.py", line 915, in set_configuration
    self._ctx.managed_set_configuration(self, configuration)
  File "D:\ProgramFiles\Python\lib\site-packages\usb\core.py", line 113, in wrapper
    return f(self, *args, **kwargs)
  File "D:\ProgramFiles\Python\lib\site-packages\usb\core.py", line 158, in managed_set_configuration
    self.managed_open()
  File "D:\ProgramFiles\Python\lib\site-packages\usb\core.py", line 113, in wrapper
    return f(self, *args, **kwargs)
  File "D:\ProgramFiles\Python\lib\site-packages\usb\core.py", line 131, in managed_open
    self.handle = self.backend.open_device(self.dev)
  File "D:\ProgramFiles\Python\lib\site-packages\usb\backend\libusb1.py", line 804, in open_device
    return _DeviceHandle(dev)
  File "D:\ProgramFiles\Python\lib\site-packages\usb\backend\_**libusb1.py", line 652, in __init__
    _check(_lib.libusb_open(self.devid, byref(self.handle)))**_
 usb.core.USBError: [Errno 13] Access denied (insufficient permissions)
```

Is usb drvier need update ?

**Comment by tormodvolden** [comment]
> usb.core.USBError: [Errno 13] Access denied (insufficient permissions)

You cannot set configuration on a device if you don't have permissions to it.

Have you assigned the device to the libusb-win32 driver using Zadig?

**Reply by Flykoki** [reply]
Have you assigned the device to the libusb-win32 driver using Zadig?  ==》 yes , use the Zading-2.9 to install libusb-win32 driver
![image](https://github.com/user-attachments/assets/105a96e0-efbf-4acb-87a9-a51bd28c00f6)

Is there any other infos need to supply for analysis this issue？

**Comment by tormodvolden** [comment]
Please provide libusb debug output. It might be easier to analyze if you test with a C program, like examples/xusb from the libusb sources, instead of using the Python module.

An lsusb dump of the device would also be nice.

You can also try not setting the configuration, or reading it out first. In some cases, if the configuration already have been set, setting it to the same value again causes a bus reset. See https://libusb.sourceforge.io/api-1.0/group__libusb__dev.html#ga785ddea63a2b9bcb879a614ca4867bed

**Comment by tormodvolden** [comment]
And please don't double-post to the mailing list as long as we are dealing with the issue here.

**Reply by Flykoki** [reply, marked as answer]
Thanks! 
After change drvier from  libusb-win32 to libusbk , issue solved!!!

**Comment by Flykoki** [comment]
libusbk working
